From df45fa02fbbe16d73717c6219c167e511bd6f7b2 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Wed, 6 Apr 2011 23:52:22 +0800
Subject: [PATCH 0065/2463] ENGR00138130-1 input: add max11801 resistance
 touchscreen controller driver

Add MAXI max11801 resistance touch screen controller driver,
This driver use Auto Mode and Aperture Mode.

Other max1180x driver can support in this driver, they use
almost same register address and define, you can find a table in
max11801_ts.c

You can find data sheet under this link:
http://www.maxim-ic.com/datasheet/index.mvp/id/5943

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 drivers/input/touchscreen/Kconfig  |   11 +++++++++++
 drivers/input/touchscreen/Makefile |    1 +
 2 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/drivers/input/touchscreen/Kconfig b/drivers/input/touchscreen/Kconfig
index 5fbc288..7cd31c0 100755
--- a/drivers/input/touchscreen/Kconfig
+++ b/drivers/input/touchscreen/Kconfig
@@ -746,4 +746,15 @@ config TOUCHSCREEN_DA9052
           Say y here to support the touchscreen found on
           Dialog Semiconductor DA9052 PMIC
 
+config TOUCHSCREEN_MAX11801
+       tristate "MAX11801 based touchscreens"
+       depends on I2C
+       help
+        Say Y here if you have a MAX11801 based touchscreen
+        controller
+
+        If unsure, say N.
+
+        To compile this driver as a module, choose M here: the
+	  module will be called max11801_ts
 endif
diff --git a/drivers/input/touchscreen/Makefile b/drivers/input/touchscreen/Makefile
index 82bb5ad..4a4dfe2 100755
--- a/drivers/input/touchscreen/Makefile
+++ b/drivers/input/touchscreen/Makefile
@@ -63,3 +63,4 @@ obj-$(CONFIG_TOUCHSCREEN_WM97XX_ZYLONITE)	+= zylonite-wm97xx.o
 obj-$(CONFIG_TOUCHSCREEN_W90X900)	+= w90p910_ts.o
 obj-$(CONFIG_TOUCHSCREEN_P1003)		+= p1003_ts.o
 obj-$(CONFIG_TOUCHSCREEN_TPS6507X)	+= tps6507x-ts.o
+obj-$(CONFIG_TOUCHSCREEN_MAX11801)	+= max11801_ts.o
-- 
1.7.7.4

